clippy_lints/src/return_self_not_must_use.rs

```diff
@@ -50,6 +51,7 @@ fn check_method(cx: &LateContext<'tcx>, decl: &'tcx FnDecl<'tcx>, fn_def: LocalD
         if decl.implicit_self.has_implicit_self();
         // We only show this warning for public exported methods.
         if cx.access_levels.is_exported(fn_def);
+        if !cx.tcx.hir().attrs(hir_id).iter().any(|attr| attr.has_name(sym::must_use));
```

Comment:
Isn't this case covered by the `must_use_attr` check in line 57?

Reply:
Indeed but apparently it's not working as it should. I'll clean up though. Good catch!

Reply:
(Which was fixed in #8143 but I realized it would be better checked here instead.)